drivers/counter/counter_nrfx_rtc.c

```diff
@@ -99,6 +99,13 @@ static int counter_nrfx_set_alarm(struct device *dev, u8_t chan_id,
 	nrfx_config->ch_data[chan_id].callback = alarm_cfg->callback;
 	nrfx_config->ch_data[chan_id].user_data = alarm_cfg->user_data;
 
+	if ((cc_val == 0) &&
+	    (get_dev_data(dev)->top != counter_get_max_top_value(dev))) {
+		/* From Product Specification: If a CC register value is 0 when
+		 * a CLEAR task is set, this will not trigger a COMPARE event.
+		 */
+		cc_val++;
```

Comment:
I'm not convinced that this is a proper way of handling this situation. The user would get this alarm one tick later than it was requested, without being aware of it.
I think it should be rather `cc_val = get_dev_data(dev)->top;`.

Reply:
As clarified offline, what I propose above won't work, as the alarm would come too early.
But I still think the driver should not silently adjust the alarm value. Maybe returning `-EINVAL` would be a better option? So that the user is aware of the situation.

Reply:
One way to detect that is that callback `ticks` parameter will be different than requested (1 instead of 0). I'm a bit reluctant when it comes to add error code as it requires handling and user i rather expecting a callback to arrive at requested time or later because it's software.

Reply:
I know, some delay is expected anyway. My point is that it would be good to notify the user that there is an attempt to achieve something that is actually impossible to achieve. The function already returns an error code (`-EINVAL` when the `ticks` value is incorrect) so the user should anyway handle it, at least with an assertion. The value of `0` would just become another one that is not permitted.

Reply:
Do you mean to return `-EINVAL` and not set it? The issue is that it's a public API and user won't know that it means that 0 is forbidden value. What can be done when user requesting ticks=0 apart from changing it to 1?

Reply:
I've added `INF` log message.

Reply:
Well, I can't see a perfect solution for this. Let's take the fair enough one that you proposed.